entviz.go

```diff
@@ -3,23 +3,83 @@ package entviz
 import (
 	"bytes"
 	_ "embed"
+	"encoding/json"
 	"html/template"
+
 	"io/ioutil"
-	"strings"
 
 	"entgo.io/ent/entc/gen"
 )
 
-//go:embed template.html
+type (
+	jsGraph struct {
+		Nodes []jsNode `json:"nodes"`
+		Edges []jsEdge `json:"edges"`
+	}
+
+	jsNode struct {
+		ID     string    `json:"id"`
+		Fields []jsField `json:"fields"`
+	}
+
+	jsEdge struct {
+		From  string `json:"from"`
+		To    string `json:"to"`
+		Label string `json:"label"`
+	}
+
+	jsField struct {
+		Name string `json:"name"`
+		Type string `json:"type"`
+	}
+)
+
+// toJsGraph converts ent's graph into json serializable struct
+func toJsGraph(g *gen.Graph) *jsGraph {
+	var jsGraph = &jsGraph{}
```

Comment:
```suggestion
	graph := &jsGraph{}
```

Avoid shadowing other types/packages.